smart_settings/smart_settings.py

```diff
@@ -3,22 +3,34 @@
 from .param_classes import NoDuplicateDict, recursive_objectify, update_recursive
 from .dynamic import recursive_dynamic_json
 import collections.abc
+
+try:
+    import tomllib
+except ImportError:
+    # tomllib was added in Python 3.11.  Older versions can use tomli
+    import tomli as tomllib
 import yaml
+
 from .utils import removesuffix
 
 IMPORT_KEY = "__import__"
 
 
 def load_raw_dict_from_file(filename):
-    """Safe load of a json file (doubled entries raise exception)"""
+    """Safe load of a json/yaml/toml file (doubled entries raise exception)."""
     if filename.endswith(".json"):
         with open(filename, "r") as f:
             data = json.load(f, object_pairs_hook=NoDuplicateDict)
-        return data
     elif filename.endswith(".yaml"):
```

Comment:
This does not fit this PR, but as you are touching the file anyways: could you also make it select YAML if the file ends on `.yml`? This has been bugging me for some time.

Reply:
added :)